crates/primitives/src/chain/spec.rs

```diff
@@ -1612,11 +1619,29 @@ struct OptimismGenesisInfo {
     regolith_time: Option<u64>,
     ecotone_time: Option<u64>,
     canyon_time: Option<u64>,
+    base_fee_params: BaseFeeParams,
 }
 
 #[cfg(feature = "optimism")]
 impl OptimismGenesisInfo {
     fn extract_from(genesis: &Genesis) -> Self {
+        let eip1559_elasticity = genesis.config.extra_fields.get("optimism").and_then(|value| {
+            value
+                .as_object()
+                .and_then(|obj| obj.get("eip1559Elasticity").and_then(|value| value.as_u64()))
+        });
+        let eip1559_denominator = genesis.config.extra_fields.get("optimism").and_then(|value| {
+            value
+                .as_object()
+                .and_then(|obj| obj.get("eip1559Denominator").and_then(|value| value.as_u64()))
+        });
+        let base_fee_params = if let (Some(elasticity), Some(denominator)) =
+            (eip1559_elasticity, eip1559_denominator)
+        {
+            BaseFeeParams::new(denominator as u128, elasticity as u128)
+        } else {
+            BaseFeeParams::ethereum()
+        };
```

Comment:
this does the "optimism" lookup twice, I think we can nest this with an if let Some